backend/test_observer/controllers/artefacts/models.py

```diff
@@ -58,6 +60,7 @@ class TestExecutionDTO(BaseModel):
     c3_link: str | None
     environment: EnvironmentDTO
     status: TestExecutionStatus
+    test_results: list[TestResult] | None
```

Comment:
I sort of understand that there is a distinction between `[]` and `None` here. As you could argue that `None` means we didn't get a report yet, while `[]` means the report has no tests. But as I see the code actually always returns `[]` so this is not really the case here. Besides it's easier to deal with a list always rather than an optional. So perhaps we can drop `None` type

Reply:
I was initially having the same view on having `None` type, but then decided it is much simpler to just return empty list if no results are available. I forgot to remove the optional `None` type here though. Thanks for flagging it!